cmd/snap-confine/ns-support.c

```diff
@@ -289,9 +290,51 @@ static bool should_discard_current_ns(dev_t base_snap_dev)
 
 enum sc_discard_vote {
 	SC_DISCARD_NO = 1,
-	SC_DISCARD_YES = 2,
+	SC_DISCARD_SHOULD = 2,
+	SC_DISCARD_MUST = 3,
 };
 
+/**
+ * is_base_transition returns true a base transition is occurring.
+ *
+ * The function inspects /run/snapd/ns/snap.$SNAP_INSTANCE_NAME.info as well
+ * as the invocation parameters of snap-confine. If the base snap name, as
+ * encoded in the info file and as described by the invocation parameters
+ * differ then a base transition is occurring. If the info file is absent or
+ * does not record the name of the base snap then transition cannot be
+ * detected.
+**/
+static bool is_base_transition(const sc_invocation * inv)
+{
+	char info_path[PATH_MAX] = { 0 };
+	sc_must_snprintf(info_path,
+			 sizeof info_path,
+			 "/run/snapd/ns/snap.%s.info", inv->snap_instance);
+
+	FILE *stream SC_CLEANUP(sc_cleanup_file) = NULL;
+	stream = fopen(info_path, "rt");
```

Comment:
`t` mode?

Reply:
Just a habit I guess. I can drop the `t` if you prefer.

Reply:
Dropped now.